### PR TITLE
Economic calendar command

### PR DESCRIPTION
**Pull request by montezdesousa**
# Description

This PR adds an economic calendar to the economy menu. The command (ecocal) output is a rich table with information regarding world economic events. The database can be sliced by date, event category, event importance, country and set a GMT time zone. User can get actual, forecast and previous data point for each event (when applicable). Historic data is available as well.

Still have to write tests but would be nice to get some feedback first, particularly on usage and defaults. I believe this data might be useful later on automated reports for e.g. It is quite relevant for any investor or economic researcher that tracks on advanced indicators and arguably accounts for a relevant part of daily market movements.

Examples (after changes):
* Lisbon time
image

* New York time
image

image

image

- [x] Summary of the change / bug fix.
- [x] Screenshot of the feature or the bug before/after fix, if applicable.
- [x] Relevant motivation and context. 

# How has this been tested?

* Please describe the tests that you ran to verify your changes. 
* Provide instructions so we can reproduce. 
* Please also list any relevant details for your test configuration.

# Checklist:

- [x] Update [our Hugo documentation](https://openbb-finance.github.io/OpenBBTerminal/) following [these guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/website).
- [x] Update our tests following [these guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/tests).
- [x] Make sure you are following our [CONTRIBUTING guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/blob/main/CONTRIBUTING.md).
- [x] If a feature was added make sure to add it to the corresponding [scripts file](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/scripts).

# Others
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My code passes all the checks pylint, flake8, black, ... To speed up development you should run `pre-commit install`.
- [x] New and existing unit tests pass locally with my changes. You can test this locally using `pytest tests/...`.

### CODE REVIEW

**Comment by minhhoang1023** [comment]
Very nice feature! 🚀 Calendar events can give some useful signals on how the economy is doing.

- When I filtered for events with a `high` level of importance, an event without any determined level of importance (aka None) still show up. It should instead be hidden.
image

- Would be nice to add a link to the event. The natural tendency is to read more about certain events to understand their implications. Right now users would need to copy-paste the event name and Google search. Some event names are rather generic, so searches can be incorrect. e.g. `Factory Orders (MoM)  (May)`

- A minor thing - there is a lot of `None` in the table. It'd be cleaner to display it as empty strings - gonna be easier to scan for information. Similar to the investing.com source
image

File: openbb_terminal/economy/economy_controller.py [hunk 1]
```diff
@@ -1000,6 +1023,98 @@ def call_ycrv(self, other_args: List[str]):
...
+            help="Event importance classified as high, medium or low.",
+        )
+        parser.add_argument(
+            "-cat",
+            "--categories",
+            action="store",
+            dest="categories",
+            nargs="+",
+            default=None,
+            help="Event category.",
```
Comment (minhhoang1023) [inline openbb_terminal/economy/economy_controller.py:1068]: you can add the param `choices` to restrict only certain values, and to show in documents the available options. Like the `interval` one below. You already added auto-completion for this purpose, but argparse wouldn't allow for autocompletion if it's not the first argument. Pain in the neck.

image

File: openbb_terminal/economy/economy_controller.py [hunk 2]
```diff
@@ -1000,6 +1023,98 @@ def call_ycrv(self, other_args: List[str]):
...
+            prog="ecocal",
+            description="Economic calendar.",
+        )
+        parser.add_argument(
+            "-tz",
+            "--timezone",
+            action="store",
+            dest="time_zone",
+            nargs="+",
+            default=None,
```
[the remark this answers is not in the document]
Comment (montezdesousa) [reply openbb_terminal/economy/economy_controller.py:1041]: yes, that makes sense

**Comment by minhhoang1023** [comment]
we should set a limit on # of events. now i had to scroll a lot get to the top...

image

**Comment by montezdesousa** [comment]
> we should set a limit on # of events. now i had to scroll a lot get to the top...
> 
> image

Did you pull the last version? 'None' should not be there and title should be "Calendar (GMT+...)" like the screen shots.

**Comment by minhhoang1023** [comment]
@montezdesousa yes that's what I mean. And we can set a default `limit` of 10. As my default is US, the return list of events is very long, spanning over several pages.

**Review by minhhoang1023 — APPROVED** [review]

**Pull request merged**